frontend/src/routes/Governance/common/util.tsx

```diff
@@ -293,39 +294,42 @@ export function resolveExternalStatus(policy: Policy) {
   return managedFields.some((mf) => knownExternalManagerPatterns.some((p) => p.test(mf.manager ?? 'none')))
 }
 
-function getHelmReleaseMap(helmReleases: HelmRelease[]) {
-  const resourceMap = new Map()
-  helmReleases.forEach((helmRelease: HelmRelease) => {
-    resourceMap.set(`${helmRelease.metadata.namespace}/${helmRelease.metadata.name}`, helmRelease)
-  })
-  return resourceMap
-}
-function getSubscriptionMap(subscriptions: Subscription[]) {
-  const resourceMap: Record<string, Subscription | undefined> = {}
-  subscriptions.forEach((subscription: Subscription) => {
-    resourceMap[`${subscription.metadata.namespace}/${subscription.metadata.name}`] = subscription
-  })
-  return resourceMap
-}
-function getChannelMap(channels: Channel[]) {
-  const channelMap: Record<string, Channel | undefined> = {}
-  channels.forEach((channel: Channel) => {
-    channelMap[`${channel.metadata.namespace}/${channel.metadata.name}`] = channel
-  })
-  return channelMap
-}
-
 // This function may need some revision/testing
-export function resolveSource(
-  policy: Policy,
+/* istanbul ignore next */
+export const resolveSource = (
+  annotations: { [key: string]: string },
   helmReleases: HelmRelease[],
   channels: Channel[],
   subscriptions: Subscription[]
-) {
+) => {
+  function getHelmReleaseMap(helmReleases: HelmRelease[]) {
+    const resourceMap = new Map()
+    helmReleases.forEach((helmRelease: HelmRelease) => {
+      resourceMap.set(`${helmRelease.metadata.namespace}/${helmRelease.metadata.name}`, helmRelease)
+    })
+    return resourceMap
+  }
+
+  function getSubscriptionMap(subscriptions: Subscription[]) {
+    const resourceMap: Record<string, Subscription | undefined> = {}
+    subscriptions.forEach((subscription: Subscription) => {
+      resourceMap[`${subscription.metadata.namespace}/${subscription.metadata.name}`] = subscription
+    })
+    return resourceMap
+  }
+
+  function getChannelMap(channels: Channel[]) {
+    const channelMap: Record<string, Channel | undefined> = {}
+    channels.forEach((channel: Channel) => {
+      channelMap[`${channel.metadata.namespace}/${channel.metadata.name}`] = channel
+    })
+    return channelMap
+  }
+
   const getAnnotations = (item: any) => item?.metadata?.annotations ?? {}
   const getHostingSubscription = (annotations: any) =>
     annotations['apps.open-cluster-management.io/hosting-subscription']
-  const parentAnnotations = getAnnotations(policy)
+  const parentAnnotations = annotations
```

Comment:
Let's just use the variable name `annotations` since `parentAnnotations` is confusing for discovered policies.